File: controller/src/main/scala/org/example/vo/Response.scala
package org.example.vo

/**
 * ${description}
 *
 * @author Lager.Tang
 *         create: 2022-04-10 19:34
 * */
case class Response(code: Int, message: String, cost: Int, req: String, result: Object) {

}

object Response {
  def success(result: Object): Response = {
    Response(200, "success", 0, null, result)
  }
}

File: interaction/src/main/java/org/example/webconfig/Response.java
package org.example.webconfig;

import lombok.Data;

/**
 * @author Lager.Tang create: 2022-04-17 13:51
 **/
@Data
public class Response<T> {

  /**
   * code 返回code
   */
  private Integer code;
  /**
   * 返回提示语
   */
  private String msg;
  /**
   * 返回结果
   */
  private T result;
  /**
   * reqId,日志索引
   */
  private String reqId;
  /**
   * 请求耗费时间
   */
  private Integer cost;


  public Response(Integer code, String msg, T result) {
    this.code = code;
    this.msg = msg;
    this.result = result;
  }

  public static <T> Response<T> success(T result) {
    return new Response<>(0, "success", result);
  }

}
